### DISCUSSION TITLE
This is just a fork of EUMETSATs work

### DISCUSSION THREAD

**Discussion by mraspaud** [pytroll/fcidecomp#2] [Announcements]
This repo is just a fork of EUMETSATs work. The pytroll community has not been involved in the development of this software, and thus does not have extensive knowledge of the code.

This fork was made in order to allow the community to make some improvements to the code, eg with updating the python versions supported by the conda recipe.

Hopefully this work will be merged back to EUMETSAT during the course of 2024.

**Comment by pdebuyl** [comment]
Hello Martin,

It is a very good idea to have this fork here. A few questions / comments:

1. There is a license: it is Apache 2.0. https://gitlab.eumetsat.int/open-source/data-tailor-plugins/fcidecomp/-/blob/2.0.1/LICENSE?ref_type=heads
2. The build scripts of EUMETSAT obfuscate a lot of things. I have found that using a "main" CMakeLists.txt simplifies the process. You can simply build as: `mkdir build ; cd build ; cmake .. ; make` for the C parts already.

This paves the way for an actual deb/rpm package.

Comment for the future: the package includes a lot of boilerplate, which I believe will end up in technical debt. The way I see it, it would be interesting to think of a plan where a fork contains only the actual source code and build items that are relevant (i.e. the CMakeLists.txt file).
